### DISCUSSION TITLE
Creating filters from object?

### DISCUSSION THREAD

**Discussion by silicahd** [alpinejs/alpine#754] [4. General]
I am running into some issues. Honestly I spend the entire day on this an I am new to Alpine but maybe there is no way to get this done.    In any case. I have an array as noted here. I am trying to create filters for a category page so i can only the the products with the filters selected. 

I am not sure if this was the correct process but what I did was to get all the products than pull all the filters into an array as seen here and than paginate the results. This way I dont end up with 2000 products inside one variable. 

This is what I am trying to get. 

Material - Wood , Stainless, Leather
Color - Brown , Gray , White
Bed Size - Queen, King

Any help is appreciate it. 

array:8 [▼
  0 => array:5 [▼
    "id" => 100
    "filter_group" => "Color"
    "product_id" => 6395
    "filter_id" => 6
    "filter_name" => "Brown"
  ]
  1 => array:5 [▼
    "id" => 101
    "filter_group" => "Material"
    "product_id" => 6395
    "filter_id" => 26
    "filter_name" => "Wood"
  ]
  2 => array:5 [▼
    "id" => 102
    "filter_group" => "Bed Size"
    "product_id" => 6395
    "filter_id" => 31
    "filter_name" => "King"
  ]
  3 => array:5 [▼
    "id" => 103
    "filter_group" => "Bed Size"
    "product_id" => 6395
    "filter_id" => 32
    "filter_name" => "Queen"
  ]
  4 => array:5 [▼
    "id" => 105
    "filter_group" => "Color"
    "product_id" => 7066
    "filter_id" => 48
    "filter_name" => "Gray"
  ]
  5 => array:5 [▼
    "id" => 106
    "filter_group" => "Material"
    "product_id" => 7066
    "filter_id" => 49
    "filter_name" => "Stainless"
  ]
  6 => array:5 [▼
    "id" => 107
    "filter_group" => "Material"
    "product_id" => 6481
    "filter_id" => 21
    "filter_name" => "Leather"
  ]
  7 => array:5 [▼
    "id" => 109
    "filter_group" => "Color"
    "product_id" => 6481
    "filter_id" => 46
    "filter_name" => "White"
  ]
]

**Comment by SimoTod** [comment]
Once you get your array with duplicates, you can do Array.from(new Set(your_array))

**Reply by silicahd** [reply]
> Once you get your array with duplicates, you can do Array.from(new Set(your_array))

Thank you for the response, Unfortunately that is not my problem. My issues is that I cant figure out how to make a drop down menu with check boxes for each filter group.

**Reply by SimoTod** [reply]
Yeah, sorry, the original question had a JS array and I didn't notice you edited it.
Can you clarify where did you get stuck? 
Are you able to reduce the original array to the one you need?
Is it a problem on your backend script or are you not sure how to use the output in Alpine?

**Reply by silicahd** [reply]
First of all thank you so much for helping me with this. I am trying to learn alpine but there seems to be very little information out there. Or maybe I can do this with alpine and I am just not understanding it. So thank you. 

At the moment I am using laravel to output the data to an api and alpine to fetch it. Now my api will push out a laravel pagination array and each product item also has its own filters. But since I am trying to oupt on the category page only the filters that products are showing in the pagination. I pulled all the filters before pagination and and than paginated the results. Giving me two arrays. One with products and pagination and one with all the filters. 

I am fairly sure this might not be the best approach, but also reduces the amount of requests on the database. 

here is my api results     https://new.istylefurniture.com/products/all?page=0

This is my code where I fetch the data. 

 `   <script>
        function products(){
            return {
                list: [], 
                page: 1,
                hideLoad: true, 
                cartOpen: false,
                isOpen: false,
                productsCount: '',
                filters: [],
                loadMore() {
                    fetch(`/products/all?page=${this.page}`)
                        .then(res => res.json())
                        .then( data =>{
                            this.list = [...this.list, ...data[0].data];
                            this.filters = data[1];
                            this.page++;
                            this.productsCount = data[0].total;
                            if(this.page > data[0].last_page){
                                this.hideLoad = false; 
                            }
                        })    
                    },
                
                }
            }
    </script>`

And this is my code where I show the data 

`<template x-for="filter in filters" :key="filter">
                       <a class="mt-3 text-gray-600 hover:underline sm:mx-3 sm:mt-0" href="#" x-text="filter.filter_group"></a>
                    </template>
`

However I am trying to accomplish something like this.  Where each filter group has its own drop down and when I select it I can show show the products from the products array. In reality it will pull another fetch as the products are paginated and filtering would be done on the laravel side.  
I guess the best question would be if I should paginate in the Laravel controller or if I should spit out all the products and paginate in alpine. The issue is that sometimes we have over 2000 products in some categories. So this is why I decided to go with the controller paginate. 

`  <template x-for="(filters ,  index) in filter" :key="index"">
         <div class="btn-group  mr-2 form-control">
          <a href="#" class="dropdown-toggle" type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" x-text="index">
          </a>
            <div class="dropdown-menu card-body p-3" style="min-width: 14rem;">
               <template x-for="(attributes as item)"">
              <label class="custom-control custom-checkbox" style="justify-content: left">
                <input type="checkbox" class="custom-control-input" x-model="tags.selected"   @click="$event.taget.value = $event.target.checked ? tags.push('item') : '' ">
                <div class="custom-control-label" x-text="item"></div>
              </label>
            </template> 
            </div>
        </div>
    </template>
 `

**Comment by silicahd** [comment]
I guess i may have figured this out. Livewire was the answer to what i was looking for. Using only alpine would have been a compromise.
